### PR TITLE
document db-manager

### PR DESCRIPTION
**Pull request by de-code**
I tried to document the current flow / sequence of calls. I am not sure whether the diagrams are useful.
In the flow diagram I tried to capture everything but it might be too confusing.
The sequence diagrams might work best (the manuscript version is good for that because it has parent and child dependencies).

To document the methods it might be best to create a base class and document it at that level rather than in each individual module. Otherwise we could have a separate document documenting the methods.

### CODE REVIEW

File: db-manager/data-import-manuscript-version.mmd [hunk 1]
```diff
@@ -0,0 +1,9 @@
+sequenceDiagram
+  main ->> dim Manuscript Version: stage_csv
+  main ->> dim Manuscript Version: applyChanges
+  dim Manuscript Version ->> dim Manuscript Version: cascadeActivations
+  dim Manuscript Version ->> dim Manuscript Version: cascadeRetirements
+  dim Manuscript Version ->> dim Manuscript: applyChanges
+  dim Manuscript Version ->> dim Manuscript: _applyChanges
```
Comment (lsh-0) [inline db-manager/data-import-manuscript-version.mmd]: is this line necessary?

**Comment by de-code** [comment]
I'm using the following command to generate the images: `docker run --rm -v "$PWD":/data:z felixlohmeier/mermaid -p -w 600 *.mmd`

**Comment by lsh-0** [comment]
cheers

File: db-manager/data-import-manuscript-version.mmd (hunk 1 above)
Comment (de-code) [reply db-manager/data-import-manuscript-version.mmd]: That was actually wrong. I corrected it, it's called on itself (I could document separately what happens in that method).

**Comment by de-code** [comment]
Do you think the flow diagram `data-import.mmd` is useful? I'm tempted to remove it if is too confusing.

(I also still need to add some docs to put the diagram into context)

**Comment by seanwiseman** [comment]
At a high level, I would say having `data-import.mmd` is better than having nothing. It could probably be paired with documentation down the road. If it's confusing then it is a good springboard for someone explaining it further.

**Comment by de-code** [comment]
Added a markdown document to give it context and also describe some of the methods.

Raised #94 to clarify some questions I wasn't sure about.

**Pull request merged**